internal/datastore/postgres/postgres.go

```diff
@@ -118,14 +146,30 @@ func NewPostgresDatastore(
 		}
 	}
 
-	return &pgDatastore{
+	gcCtx, cancelGc := context.WithCancel(context.Background())
+
+	datastore := &pgDatastore{
 		dburl:                     url,
 		dbpool:                    dbpool,
 		watchBufferLength:         config.watchBufferLength,
 		revisionFuzzingTimedelta:  config.revisionFuzzingTimedelta,
 		gcWindowInverted:          -1 * config.gcWindow,
+		gcInterval:                config.gcInterval,
+		gcMaxOperationTime:        config.gcMaxOperationTime,
 		splitAtEstimatedQuerySize: config.splitAtEstimatedQuerySize,
-	}, nil
+		gcCtx:                     gcCtx,
+		cancelGc:                  cancelGc,
+	}
+
+	// Start a goroutine for garbage collection.
+	if datastore.gcInterval > 0*time.Minute {
```

Comment:
i think there's a time.IsZero we could use here `if !time.IsZero(datastore.gcInterval)`